statsd/statsd.go

```diff
@@ -70,17 +70,13 @@ const (
 
 /*
 ddEnvTagsMapping is a mapping of each "DD_" prefixed environment variable
-to a specific tag name.
+to a specific tag name. We use a slice to keep the order and simplify tests.
 */
-var ddEnvTagsMapping = map[string]string{
-	// Client-side entity ID injection for container tagging.
-	"DD_ENTITY_ID": "dd.internal.entity_id",
-	// The name of the env in which the service runs.
-	"DD_ENV": "env",
-	// The name of the running service.
-	"DD_SERVICE": "service",
-	// The current version of the running service.
-	"DD_VERSION": "version",
+var ddEnvTagsMapping = [][]string{
+	[]string{"DD_ENTITY_ID", "dd.internal.entity_id"}, // Client-side entity ID injection for container tagging.
+	[]string{"DD_ENV", "env"},                         // The name of the env in which the service runs.
+	[]string{"DD_SERVICE", "service"},                 // The name of the running service.
+	[]string{"DD_VERSION", "version"},                 // The current version of the running service.
```

Comment:
```suggestion
var ddEnvTagsMapping = []struct{envName, tagName}{
	{"DD_ENTITY_ID", "dd.internal.entity_id"}, // Client-side entity ID injection for container tagging.
	{"DD_ENV", "env"},                         // The name of the env in which the service runs.
	{"DD_SERVICE", "service"},                 // The name of the running service.
	{"DD_VERSION", "version"},                 // The current version of the running service.
```

And maybe const too?